<launch>
  <arg name="mn" default=""/>
  <arg name="mc_x" default="0"/>
  <arg name="mc_y" default="0"/>
  <arg name="mb_w" default="0"/>
  <arg name="mb_h" default="0"/>
  <arg name="ts"/>
  <arg name="sp_x"/>
  <arg name="sp_y"/>
  <arg name="pn" default=""/>
  <arg name="lv"/>
  <arg name="pav"/>
  <arg name="nav"/>
  <arg name="l_cr"/>
  <arg name="l_af"/>
  <arg name="e_rd"/>
  <arg name="e_md"/>
  <arg name="e_p"/>
  <arg name="d_lp" default="0"/>
  <arg name="d_ap" default="0"/>
  <arg name="t_lsc" default="0"/>
  <arg name="t_asc" default="0"/>
  <arg name="d" default="0"/>

  <node pkg="wandrian" type="run" name="run" output="screen">
    <remap from="run/power" to="mobile_base/commands/motor_power"/>
    <remap from="run/velocity" to="mobile_base/commands/velocity"/>
    <remap from="run/odometry" to="odom"/>
    <remap from="run/laser" to="scan"/>

    <param name="map_name" value="$(arg mn)" type="string"/>
    <param name="map_center_x" value="$(arg mc_x)" type="double"/>
    <param name="map_center_y" value="$(arg mc_y)" type="double"/>
    <param name="map_boundary_width" value="$(arg mb_w)" type="double"/>
    <param name="map_boundary_height" value="$(arg mb_h)" type="double"/>
    <param name="tool_size" value="$(arg ts)" type="double"/>
    <param name="starting_point_x" value="$(arg sp_x)" type="double"/>
    <param name="starting_point_y" value="$(arg sp_y)" type="double"/>
    <param name="plan_name" value="$(arg pn)" type="string"/>
    <param name="linear_velocity" value="$(arg lv)" type="double"/>
    <param name="positive_angular_velocity" value="$(arg pav)" type="double"/>
    <param name="negative_angular_velocity" value="$(arg nav)" type="double"/>
    <param name="laser_count_rate" value="$(arg l_cr)" type="double"/>
    <param name="laser_augmentation_factor" value="$(arg l_af)" type="double"/>
    <param name="epsilon_rotational_direction" value="$(arg e_rd)" type="double"/>
    <param name="epsilon_motional_direction" value="$(arg e_md)" type="double"/>
    <param name="epsilon_position" value="$(arg e_p)" type="double"/>
    <param name="deviation_linear_position" value="$(arg d_lp)" type="double"/>
    <param name="deviation_angular_position" value="$(arg d_ap)" type="double"/>
    <param name="threshold_linear_step_count" value="$(arg t_lsc)" type="double"/>
    <param name="threshold_angular_step_count" value="$(arg t_asc)" type="double"/>
    <param name="delay" value="$(arg d)" type="double"/>

    <param name="linear_velocity_step" value="0.15" type="double"/>
    <param name="linear_velocity_max" value="1.5" type="double"/>
    <param name="angular_velocity_step" value="0.75" type="double"/>
    <param name="angular_velocity_max" value="3" type="double"/>
  </node>
</launch>
